crates/test-framework/src/spicetest/vector_search/evaluate.rs

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::collections::HashMap;
+
+/// Evaluate the search results using NDCG@10 metric.
+/// Reference: `https://github.com/embeddings-benchmark/mteb/blob/03347ebfe4809056e0fd2894fcae69dcdd2ed964/mteb/evaluation/evaluators/RetrievalEvaluator.py#L500`
+#[allow(clippy::cast_precision_loss)]
+#[must_use]
+pub(crate) fn evaluate_search_results<S: ::std::hash::BuildHasher>(
```

Comment:
Existing code moved under test-framework. Original code will be removed after full migration
https://github.com/spiceai/spiceai/blob/d4683b76e36acab79e65379bc721fa7eecd2ecc8/crates/runtime/benches/bench_search/evaluator.rs#L22